R/Find_MisSegregationRate_Surrogate.R

```diff
@@ -62,24 +63,70 @@ allmat = list()
 paths = list()
 for (can in cancerList){
   print(paste0("working on ",can))
-  X <- curatedTCGAData(diseaseCode = can, assays = "RNASeq2*", dry.run = FALSE)
+  X <- curatedTCGAData(diseaseCode = can, assays = "RNASeq2*", dry.run = FALSE, version = "2.0.1")
   # sampleTables(X)
-  tmp = splitAssays(X, c("01"))
-  Y = tmp[[grep(paste0("01_",can,"_RNASeq2"),
-                names(tmp))]]@assays$data@listData[[1]]
-  Y = Y[,1:min(100,ncol(Y))]
+  # tmp = splitAssays(X, sampleTables(X))
+  # Y = tmp[[grep(paste0("01_",can,"_RNASeq2"),
+  #               names(tmp))]]@assays$data@listData[[1]]
+  # tmp=as.matrix(assays(X)[["HNSC_RNASeq2GeneNorm-20160128"]])
+  tmp=as.matrix(assays(X)[[2]])
+  sampleType=sapply(colnames(tmp), function(x) strsplit(x, "-")[[1]][4])
+  tmp=tmp[,grep("01",sampleType)]
+  Y = tmp[,1:min(20,ncol(tmp))]
   allmat[[can]] = Y
 }
 
+# tmp=as.matrix(assays(X)[["HNSC_RNASeq2GeneNorm-20160128"]])
+# sampleType=sapply(colnames(tmp), function(x) strsplit(x, "-")[[1]][4])
+# tmp=tmp[,grep("01",sampleType)]
+
+###############################################################################
+### Integrate TCGA & GEO data adjusting for Batch Effects using comBat-seq ####
+###############################################################################
+combined<-data.frame(ncol=1)
+for (name in names(allmat)){
+  combined=cbind(combined, allmat[[name]])
+}
+combined = cbind(combined, Z[rownames(allmat$HNSC),])
+combined <- combined[,-c(1)]
+combined =as.matrix(combined)
+batch <- c(rep(1, 20), rep(2, 20),rep(3, 20),rep(4, 20),rep(5, 20),rep(6, 20),rep(7, 20),rep(8, 20), rep(9, 20), rep(10, 39))
```

Comment:
`sapply(allmat,ncol)` should give u the second argument of `rep` here without hard coding

Reply:
Got it in f5caf2b